x-pack/platform/plugins/shared/entity_manager/server/routes/v2/count.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { z } from '@kbn/zod';
+import { createEntityManagerServerRoute } from '../create_entity_manager_server_route';
+import { READ_ENTITIES_PRIVILEGE } from '../../lib/v2/constants';
+
+export const countEntitiesRoute = createEntityManagerServerRoute({
+  endpoint: 'POST /internal/entities/v2/_count',
+  security: {
+    authz: {
+      requiredPrivileges: [READ_ENTITIES_PRIVILEGE],
+    },
+  },
+  params: z.object({
+    body: z.object({
+      types: z.optional(z.array(z.string())).default([]),
+      filters: z.optional(z.array(z.string())).default([]),
+      start: z
+        .optional(z.string())
+        .default(() => moment().subtract(5, 'minutes').toISOString())
+        .refine((val) => moment(val).isValid(), {
+          message: '[start] should be a date in ISO format',
+        }),
+      end: z
+        .optional(z.string())
+        .default(() => moment().toISOString())
+        .refine((val) => moment(val).isValid(), {
+          message: '[end] should be a date in ISO format',
+        }),
```

Comment:
Since we pass these params to the client, perhaps it's worth extracting this RT into the types file and re-using the type contract in both places? At least until the route needs to be different from the client interface.

Reply:
https://github.com/elastic/kibana/pull/203605/commits/e5ef75f90582a5cd6a6484456f4fb8f8eaf4cbc4